tortoise/ninja_tortoise.go

```diff
@@ -490,35 +499,21 @@ func (ni *NinjaTortoise) updatePatSupport(p votingPattern, bids []types.BlockID,
 	ni.TPatSupport[p][idx] = votingPattern{id: pid, LayerID: idx}
 }
 
-func initTallyToBase(tally map[votingPattern]map[types.BlockID]vec, base votingPattern, p votingPattern) {
-	tally[p] = make(map[types.BlockID]vec)
-	for k, v := range tally[base] {
-		tally[p][k] = v
+func (ni *NinjaTortoise) initTallyToBase(base votingPattern, p votingPattern, windowStart types.LayerID) {
+	mp := make(map[BlockIDLayerTuple]vec, len(ni.TTally[base]))
+	for b, v := range ni.TTally[base] {
+		if b.Layer() < windowStart {
+			continue //dont copy votes for block outside window
+		}
+		mp[b] = v
 	}
+	ni.TTally[p] = mp
 }
 
 func (ni *NinjaTortoise) latestComplete() types.LayerID {
 	return ni.PBase.Layer()
 }
 
-func (ni *NinjaTortoise) getVotes() map[types.BlockID]vec {
-	return ni.TVote[ni.PBase]
-}
-
-func (ni *NinjaTortoise) getVote(id types.BlockID) vec {
-	block, err := ni.GetBlock(id)
-	if err != nil {
-		ni.Panic(fmt.Sprintf("error block not found ID %s, %v", id, err))
-	}
-
-	if block.Layer() > ni.PBase.Layer() {
-		ni.Error("we dont have an opinion on block according to current pbase")
-		return Against
-	}
-
-	return ni.TVote[ni.PBase][id]
-}
-
```

Comment:
Aren't we using these functions at all?
Why they were here at first place?

Reply:
metrics, these ones where wrong and not used